opentelemetry-sdk/src/opentelemetry/sdk/trace/export/__init__.py

```diff
@@ -164,60 +169,128 @@ def on_end(self, span: Span) -> None:
 
```

Comment:
Can this notify still be missed?

Reply:
it might be missed but i don't think it is a problem. 
missing the notificiation call here means that the worker thread is currently not waiting but running an export operation which reduces the number of spans in the queue. the worker thread will only got to sleep if the current number of spans is below the size of one batch. so missing the notification call should not be a problem since the export would start right away once the currently running export operation is finished.